server/services/v1/search/session.go

```diff
@@ -64,10 +64,25 @@ func (sessions *SessionManager) Execute(ctx context.Context, runner Runner) (Res
 		return Response{}, errors.NotFound("tenant '%s' not found", namespace)
 	}
 
-	resp, err := runner.Run(ctx, tenant)
+	resp, err := sessions.execute(ctx, tenant, runner)
+	if err != nil && shouldRecheckTenantVersion(err) {
+		_ = sessions.TrackVersion(ctx, tenant)
+
+		resp, err = sessions.execute(ctx, tenant, runner)
```

Comment:
Can this again return `metadata.ErrCodeProjectNotFound, metadata.ErrCodeSearchIndexNotFound` ?

Reply:
that's right and the `createApiError` at line 74 will wrap it up in user facing error.

Reply:
Then should we run this in the loop, something like:
```
do {
 resp, err := sessions.execute(ctx, tenant, runner)
} while	(err != nil && shouldRecheckTenantVersion(err))
```

Reply:
We only need to check & reload once.